cypress/integration/core_opensearch_dashboards/opensearch_dashboards/apps/query_enhancements/table.spec.js

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  INDEX_WITH_TIME_1,
+  INDEX_WITHOUT_TIME_1,
+  SECONDARY_ENGINE,
+} from '../../../../../utils/constants';
+import {
+  getRandomizedWorkspaceName,
+  getRandomizedDatasourceName,
+  generateAllTestConfigurations,
+  generateIndexPatternTestConfigurations,
+  setDatePickerDatesAndSearchIfRelevant,
+} from '../../../../../utils/apps/query_enhancements/shared';
+import { QueryLanguages } from '../../../../../utils/apps/query_enhancements/constants';
+import { selectFieldFromSidebar } from '../../../../../utils/apps/query_enhancements/sidebar';
+
+const workspaceName = getRandomizedWorkspaceName();
+const datasourceName = getRandomizedDatasourceName();
+
+const generateTableTestConfiguration = (dataset, datasetType, language) => {
+  const baseConfig = {
+    dataset,
+    datasetType,
+    language: language.name,
+    testName: `${language.name}-${datasetType}`,
+  };
+
+  return {
+    ...baseConfig,
+  };
+};
+
+export const runTableTests = () => {
+  describe('discover table tests', () => {
+    beforeEach(() => {
+      cy.setupTestData(
+        SECONDARY_ENGINE.url,
+        [
+          `cypress/fixtures/query_enhancements/data_logs_1/${INDEX_WITH_TIME_1}.mapping.json`,
+          `cypress/fixtures/query_enhancements/data_logs_1/${INDEX_WITHOUT_TIME_1}.mapping.json`,
+        ],
+        [
+          `cypress/fixtures/query_enhancements/data_logs_1/${INDEX_WITH_TIME_1}.data.ndjson`,
+          `cypress/fixtures/query_enhancements/data_logs_1/${INDEX_WITHOUT_TIME_1}.data.ndjson`,
+        ]
+      );
+      cy.addDataSource({
+        name: datasourceName,
+        url: SECONDARY_ENGINE.url,
+        authType: 'no_auth',
+      });
+      //cy.deleteWorkspaceByName(`${workspaceName}`);
+      cy.visit('/app/home');
+      cy.osd.createInitialWorkspaceWithDataSource(datasourceName, workspaceName);
+    });
+
+    afterEach(() => {
+      cy.deleteWorkspaceByName(`${workspaceName}`);
+      cy.deleteDataSourceByName(datasourceName);
+      cy.deleteIndex('data_logs_small_time_1');
+      cy.window().then((win) => {
+        win.localStorage.clear();
+        win.sessionStorage.clear();
+      });
+    });
+
+    generateAllTestConfigurations(generateTableTestConfiguration, {
+      indexPattern: 'data_logs_small_time_1*',
+      index: 'data_logs_small_time_1',
+    }).forEach((config) => {
+      describe(`${config.testName}`, () => {
+        beforeEach(() => {
+          if (config.datasetType === 'INDEX_PATTERN') {
```

Comment:
nitpick: `config.datasetType === DataTypes.IndexPattern.name` (or something like that. i just hand wrote this)